File: 108_Convert_Sorted_Array_to_Binary_Search_Tree/108_1.c
struct TreeNode *subset(int *nums, int left, int right) {
    if (right < left) return NULL;
    int mid = (left + right) >> 1;
    struct TreeNode *root = (struct TreeNode *)malloc(sizeof(struct TreeNode));
    root->val = nums[mid];
    
        root->left  = subset(nums, left, mid - 1);
        root->right = subset(nums, mid + 1, right);
    
    return root;
}

struct TreeNode* sortedArrayToBST(int* nums, int numsSize){
    if (numsSize == 0) return NULL;
    return subset(nums, 0, numsSize - 1);
}

File: 108_Convert_Sorted_Array_to_Binary_Search_Tree/108_1.py
from typing import List

class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        def build_tree(nums, lo, hi) -> TreeNode:
            if hi < lo: 
                return
            mid = (lo + hi) >> 1
            root = TreeNode(nums[mid])
            root.left  = build_tree(nums, lo, mid - 1)
            root.right = build_tree(nums, mid + 1, hi)
            return root

        return build_tree(nums, 0, len(nums) - 1)